Lib/test/test_pdb.py

```diff
@@ -4331,15 +4335,38 @@ def test_quit(self):
 
         stdout, stderr = self._run_script(script, commands)
         self.assertIn("2", stdout)
-        self.assertIn("Quit anyway", stdout)
+        self.assertIn(QUIT_PROMPT_QUESTION, stdout)
         # Closing stdin will quit the debugger anyway so we need to confirm
         # it's the quit command that does the job
         # call/return event will print --Call-- and --Return--
         self.assertNotIn("--", stdout)
         # Normal exit should not print anything to stderr
         self.assertEqual(stderr, "")
         # The quit prompt should be printed exactly twice
-        self.assertEqual(stdout.count("Quit anyway"), 2)
+        self.assertEqual(stdout.count(QUIT_PROMPT_QUESTION), 2)
+
+
+@support.force_not_colorized_test_class
+@support.requires_subprocess()
+class TestREPLSession(unittest.TestCase):
+    def test_return_from_inline_mode_to_REPL(self):
+        # GH-124703: Raise BdbQuit when exiting pdb in REPL session.
+        # This allows the REPL session to continue.
+        user_input = """
```

Comment:
Personally I would move this after `p = spawn_repl()`. If we need to do a local import I would rather do it at the beginning of the function, like we do at module levels.